File: 0x07-rotate_2d_matrix/0-rotate_2d_matrix.js
function rotate_2d_matrix(matrix) {
    const c = matrix.length;

    for (let i = 0; i < c; i++) {
        for (let j = i; j < c; j++) {
            let temp = matrix[i][j];
            matrix[i][j] = matrix[j][i];
            matrix[j][i] = temp;
        }
    }

    for (let i = 0; i < c; i++) {
        matrix[i].reverse();
    }
}

const matrix = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
];
rotate_2d_matrix(matrix);
console.log(matrix);

File: 0x07-rotate_2d_matrix/0-rotate_2d_matrix.py
#!/usr/bin/python3
""" Rotate 2D Matrix
"""

def rotate_2d_matrix(matrix):
    """ Given an n x n 2D matrix, rotate it 90 degrees clockwise
    """
    newMatrix = zip(*reversed(matrix))
    for n, x in enumerate(newMatrix):
        matrix[n] = list(x)
